src/ranged.cpp

```diff
@@ -1200,9 +1200,11 @@ std::vector<tripoint> target_handler::target_ui( player &pc, target_mode mode,
         pc.cancel_activity();
 
         tripoint targ( 0, 0, 0 );
+        cata::optional<tripoint> mouse_pos;
         // Our coordinates will either be determined by coordinate input(mouse),
         // by a direction key, or by the previous value.
-        if( action == "SELECT" && ctxt.get_coordinates( g->w_terrain, targ.x, targ.y ) ) {
+        if( action == "SELECT" && ( mouse_pos = ctxt.get_coordinates( g->w_terrain ) ) ) {
```

Comment:
Is it valid syntax? You have used result of assignment as condition. Will it return true only if assignment was successful?

Reply:
`mouse_pos = ctxt.get_coordinates( g->w_terrain )` first assigns result of `ctxt.get_coordinates( g->w_terrain )` to `mouse_pos`, then evaluates to the value of `mouse_pos`.
Type was `bool` now `cata::optional<tripoint>` with `operator bool()` defined.